plugins/system_upgrade.py

```diff
@@ -680,7 +686,7 @@ def transaction_upgrade(self):
                         UPGRADE_FINISHED_ID)
         self.run_clean()
         if self.opts.tid[0] == "upgrade":
-            reboot()
+            poweroff()
```

Comment:
Definitely not unconditionally, sorry.

I guess an option for this would make sense. You'd have to store it in the state file when preparing the data before reboot, then read it here and either reboot or shutdown.

Reply:
Hey lukash,

Thank you for your feedback. Perhaps we can make it a configuration option in `/etc/dnf/plugins/system-upgrade.conf`? That way you configure it once and never think about it again, and it would also work when invoked through a GUI that doesn't know about this option.

Reply:
Well, my point kind of is that you want to be explicit about it. You don't normally do a `system-upgrade` more than once in 6 months, so it would save you typing additional `--poweroff` twice a year (multiplied by the number of machines you maintain).

Not that it would be such an issue to have it in the configuration, it just doesn't feel like the best place for it.

There is no GUI for the `system-upgrade` plugin that I know of? It's a command-line tool.

Also note the plugins don't share their own configuration options with DNF and the `system-upgrade` plugin has no configuration file as of yet, so you'd need to add that. Doesn't seem worth it for this particular option.

Still seems to me a `--poweroff` command line switch would be the best solution for you.

Reply:
Hey lukash,

Well, there's a way to do Fedora  system upgrades with GNOME Software:
https://docs.fedoraproject.org/en-US/quick-docs/upgrading/#sect-upgrading-to-the-next-fedora-workstation-release
I assumed that it would use the system-upgrade plugin behind the scenes, but apparently I was wrong.

Reply:
@lukash,

The reason I'd like to configure this through a configuration file is that by the time that I perform my next system upgrade I'll probably forget that the flag even exists, not to mention what it's called (“was it `--shutdown`? Or `--shut-down`? `--poweroff`?”). In my opinion, whether you want the machine to shut down or reboot after a system upgrade is something that is likely to stay the same over the lifetime of the machine. You're running it on a server? You probably always want it to reboot. On a private PC? You probably always want it to shut down. So to me, this definitely feels like something that belongs in a configuration file, not on a command line flag.

Reply:
Well, if you decide to run the upgrade late in the evening and you know you want the machine to shut down, you can run `--help` to find out what the command line switch is and use it.

If you're doing it through the GUI, there could be like a check box... I don't use that, so I don't know how it looks. Anyway, someone would have to implement it for GNOME Software and I don't think any support is needed from DNF, it would just shut down after a (successful?) upgrade.

Anyway, is there something more to be done in this PR or shall we close it?